agent/agent_test.go

```diff
@@ -95,4 +95,85 @@ func TestAgentStartStop(t *testing.T) {
 	}
 
 	assert.NoError(t, agent.Stop(ctx))
+	ready := agent.Ready()
+
+	if ready != nil {
+		t.Fatalf("ready cannot return nil channel")
+	}
+	agent.Start(ctx)
+
+	ready = agent.Ready()
+	if ready != nil {
+		t.Fatalf("ready cannot be nil")
+	}
+
+}
+
+func TestHandleSessionMessage(t *testing.T) {
+	ctx, _ := context.WithTimeout(context.Background(), 5000*time.Millisecond)
+	agent, tc, cleanup := agentTestEnv(t)
+	defer tc.Stop()
```

Comment:
Ok, sorry I wasn't clear before. Go look at the implementation of `storageTestEnv`. It puts all of the cleanup functions in to a single cleanup function that is returned. `tc` isn't even relevant to this test.